core/src/main/java/org/wildfly/extras/creaper/core/online/ManagementProtocol.java

```diff
@@ -10,11 +10,15 @@
     /**
      * HTTP transport is used. Default port 9990
      * Asynchronous execution is not provided.
-     * If an error occurs (server does not respond 401 or header does not contain WWW-Authenticate field after
-     * first request) {@code IllegalStateException is thrown}
      * Execution on is allowed with <b>no attachments</b> (i.e. deployment operation).
      */
     HTTP("http"),
+    /**
+     * HTTPS transport is used. Default port 9993
+     * Asynchronous execution is not provided.
+     * Execution on is allowed with <b>no attachments</b> (i.e. deployment operation).
+     */
+    HTTPS("https"),
```

Comment:
Aha, OK.

I was thinking that maybe there should be no `HTTPS` nor `HTTPS_REMOTING` in `ManagementProtocol`, only `HTTP` and `HTTP_REMOTING`, and it would be switched to the `S` variant if `SslOptions` are defined, but that's probably a bit too magical.

Default `SSLContext` looks good. But please don't use `org.apache.http.conn.ssl.SSLContexts` if it's deprecated and points to a replacement: `org.apache.http.ssl.SSLContexts`. And `SSLContexts.createSystemDefault` sounds good, but maybe there would be unnecessary difference between `HTTPS` and `HTTPS_REMOTING` then. I think these two should be consistent in how they configure the `SSLContext`.